### PR TITLE
CVE-2002-0661 path traversal using backslash

### PR DESCRIPTION
**Pull request by renchauret**

### CODE REVIEW

File: cves/CVE-2002-0661.yml [hunk 1]
```diff
@@ -87,17 +97,20 @@ discovered:
     originally found. Answer in longform below in "answer", fill in the date in
     YYYY-MM-DD, and then determine if the vulnerability was found by a Google
     employee (you can tell from their email address). If it's clear that the
-    vulenrability was discovered by a contest, fill in the name there.
+    vulnerability was discovered by a contest, fill in the name there.
```
Comment (randallkentbrown) [inline cves/CVE-2002-0661.yml:100]: Good job finding this typo.

File: cves/CVE-2002-0661.yml [hunk 2]
```diff
@@ -41,19 +50,18 @@ bounty:
   announced:
```
Comment (randallkentbrown) [inline cves/CVE-2002-0661.yml:50]: Was there no information online announcing the vulnerability before the fix was published?

File: cves/CVE-2002-0661.yml [hunk 3]
```diff
@@ -118,10 +131,14 @@ interesting_commits:
...
-    - commit:
-      note:
+    - commit: 238a9d6e51ba186e6903272c16a6ec695e3654d6
+      note: |
+        Much development occurred on mod_negotiation.c, the primary file containing 
+        the vulnerability, but little was relevant to the vulnerability as the file 
+        is quite large. This is the only commit which works with this area of the code. 
+        It adds a comment which illustrates that the author considered path traversal 
+        and the current way it is handled but did not consider that checking for '/' 
+        would not be sufficient.
```
Comment (randallkentbrown) [inline cves/CVE-2002-0661.yml]: Potentially change this last line to "would be insufficient."

File: cves/CVE-2002-0661.yml [hunk 4]
```diff
@@ -199,4 +224,13 @@ mistakes:
...
-  answer:
+  answer: |
+    The code mistake made here was only checking for '/' when sanitizing a 
+    potential path traversal vulnerability. If I try to put myself into the 
+    minds of the developers, I imagine they, likely with mostly UNIX experience, 
+    simply did not consider that backslashes could be used for path traversal 
+    in other operating systems. Even one developer who looked at and commented 
+    on the code in question between the VCC and the fix did not catch the issue 
+    before release. This shows the importance of thoroughly testing and viewing 
+    your software from different perspectives--perspectives which might reveal
```
Comment (randallkentbrown) [inline cves/CVE-2002-0661.yml]: I would write this as "... your software from different perspectives, which might reveal new problems."

File: cves/CVE-2002-0661.yml (hunk 3 above)
Comment (renchauret) [reply cves/CVE-2002-0661.yml]: yeah that's a bit more concise

File: cves/CVE-2002-0661.yml (hunk 4 above)
Comment (renchauret) [reply cves/CVE-2002-0661.yml]: also more concise, will do

File: cves/CVE-2002-0661.yml (hunk 2 above)
Comment (renchauret) [reply cves/CVE-2002-0661.yml:50]: this is the announced line under the bounty section (which there was none), not the announced line for the vulnerability

Comment (randallkentbrown) [reply cves/CVE-2002-0661.yml:50]: Totally correct, my bad!

File: cves/CVE-2002-0661.yml [hunk 5]
```diff
@@ -107,8 +120,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer:
-  name:
+  answer: Based on filepaths and bug report description.
```
Comment (based-avocado) [inline cves/CVE-2002-0661.yml]: May want to give a little more detail here, like a path to the corresponding directory or something. Not 100% necessary, but this answer struck me as a bit sparse

File: cves/CVE-2002-0661.yml [hunk 6]
```diff
@@ -131,12 +148,16 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer:
+  answer: |
+    During the time of the vulnerability's existence, the development team was
```
Comment (based-avocado) [inline cves/CVE-2002-0661.yml]: Maybe change this to "While this vulnerability was present in the system" or something like that. First sentence is currently a little clunky

File: cves/CVE-2002-0661.yml [hunk 7]
```diff
@@ -199,4 +224,13 @@ mistakes:
...
+  answer: |
+    The code mistake made here was only checking for '/' when sanitizing a 
+    potential path traversal vulnerability. If I try to put myself into the 
+    minds of the developers, I imagine they, likely with mostly UNIX experience, 
+    simply did not consider that backslashes could be used for path traversal 
+    in other operating systems. Even one developer who looked at and commented 
+    on the code in question between the VCC and the fix did not catch the issue 
+    before release. This shows the importance of thoroughly testing and viewing 
+    your software from different perspectives--perspectives which might reveal 
+    new problems.
```
Comment (based-avocado) [inline cves/CVE-2002-0661.yml]: Consider slapping a newline onto the end of this file: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: cves/CVE-2002-0661.yml (hunk 5 above)
Comment (renchauret) [reply cves/CVE-2002-0661.yml]: Alright, I'll elaborate on the filepath

File: cves/CVE-2002-0661.yml (hunk 6 above)
Comment (renchauret) [reply cves/CVE-2002-0661.yml]: Agreed

**Comment by based-avocado** [comment]
Upvotes: 2

File: cves/CVE-2002-0661.yml (hunk 5 above)
Comment (based-avocado) [reply cves/CVE-2002-0661.yml]: Cheers. That'll add a nice little bit of context for how you came up with the subsystem name

**Comment by randallkentbrown** [comment]
3 upvotes

**Pull request merged**